inc/saiport.h

```diff
@@ -236,6 +236,24 @@ typedef enum _sai_port_interface_type_t
 
     /** Interface type KR4 */
     SAI_PORT_INTERFACE_TYPE_KR4,
+
+    /** Interface type CAUI */
+    SAI_PORT_INTERFACE_TYPE_CAUI,
+
+    /** Interface type GMII */
+    SAI_PORT_INTERFACE_TYPE_GMII,
+
+    /** Interface type SFI */
+    SAI_PORT_INTERFACE_TYPE_SFI,
+
+    /** Interface type XLAUI */
+    SAI_PORT_INTERFACE_TYPE_XLAUI,
+
```

Comment:
@kcudnik Do we need to add SAI_PORT_INTERFACE_TYPE_MAX or extensions can be added using CUSTOM_RANGE_START.

Reply:
how do we plan to use SAI_PORT_INTERFACE_TYPE_MAX?